### ISSUE TITLE
[a11y-defect-2]: Your VA Payments loses keyboard focus when the React app finishes loading

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#95239]
### Point of contact

Trevor Pierce

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The Your VA Payments `va-payment-history/payments/` view has a React app nested inside the static container. Focus is lost to the `<body>` tag when the React app finishes loading. This creates a situation where keyboard users are unsure where focus will go when they press `TAB` and screen reader users do not hear the app being loaded. Screenshot attached below.

---

![Image](https://github.com/user-attachments/assets/42c7d348-8ddc-4621-bc93-90e555b4ec74)

### Reproduction steps

1. Log into `va-payment-history/` with a test user who has one or more payments in their history
2. Turn on the screen reader of your choice
3. Click the "View your VA payment history" link to load the history table view
4. Verify the screen reader does not announce the route change or the page being loaded. This was confirmed with NVDA on Chrome and Firefox, Win11.

### Proposed solution or next steps

Keyboard (and by proxy screen reader) focus must be actively managed. In this case, I feel it would be best to set focus on the H1 when the entire React application is loaded and in the ready state. This will require the H1 have a `tabindex="-1"` added so it can receive focus programmatically.

### References, articles, or WCAG support

1. Understanding [SC 1.3.2](https://w3.org/TR/WCAG22#meaningful-sequence): Meaningful Sequence (Level A)
2. Understanding [SC 1.3.3](https://w3.org/TR/WCAG22#sensory-characteristics): Sensory Characteristics (Level A)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [x] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [x] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by dysbo** [comment]
Refinement notes:
- should be able to use `useEffect`
    - check for `ref.current` and if it exists, `ref.current.focus()` with empty dependency array so it fires only on initial page load
- create a `ref`
- assign the `ref` to the `H1`
- add `tabindex="-1"` to the `H1`

**Comment by salihaghaffar** [comment]
@1Copenut  can you help verify this, for me the focus is not going to Your VA Payments.

**Comment by 1Copenut** [comment]
@salihaghaffar I tested this PR in the Staging environment just now. NVDA announces the heading properly and gives a good indication to users that the table has loaded.

NVDA is a good proxy for JAWS and other screen readers using this technique. I've used it in a number of cases and had consistent results with NVDA, JAWS, and VoiceOver.
